engines/private/funcs.cpp

```diff
@@ -0,0 +1,784 @@
+    debugC(1, kPrivateDebugScript, "WARNING: DossierChgSheet is not implemented");
+}
+
+void fDossierPrevSuspect(ArgArray args) {
+    assert (args.size() == 3);
+    Common::String *s = new Common::String(args[0].u.str);
+    MaskInfo *m = (MaskInfo *)malloc(sizeof(MaskInfo));
+
+    int x = args[1].u.val;
+    int y = args[2].u.val;
+
+    m->surf = g_private->loadMask(*s, x, y, true);
+    m->cursor = new Common::String("kExit");
+    m->nextSetting = NULL;
+    m->flag1 = NULL;
+    m->flag2 = NULL;
+    g_private->_dossierPrevSuspectMask = m;
+    g_private->_masks.push_front(*m);
+}
+
+void fDossierNextSuspect(ArgArray args) {
+    assert (args.size() == 3);
+    Common::String *s = new Common::String(args[0].u.str);
+    MaskInfo *m = (MaskInfo *)malloc(sizeof(MaskInfo));
+
+    int x = args[1].u.val;
+    int y = args[2].u.val;
+
+    m->surf = g_private->loadMask(*s, x, y, true);
+    m->cursor = new Common::String("kExit");
+    m->nextSetting = NULL;
+    m->flag1 = NULL;
+    m->flag2 = NULL;
+    g_private->_dossierNextSuspectMask = m;
+    g_private->_masks.push_front(*m);
+}
+
+void fNoStopSounds(ArgArray args) {
+    assert(args.size() == 0);
+    debugC(1, kPrivateDebugScript, "NoStopSounds()");
+    g_private->_noStopSounds = true;
+}
+
+void fLoseInventory(ArgArray args) {
+    assert(args.size() == 0);
+    debugC(1, kPrivateDebugScript, "LoveInventory()");
+    g_private->inventory.clear();
+}
+
+void fInventory(ArgArray args) {
+    // assert types
+    Datum b1 = args[0];
+    Datum v1 = args[1];
+    Datum v2 = args[2];
+    Datum e = args[3];
+    Datum i = args[4];
+    Datum c = args[5];
+    Datum snd = args[8];
+
+    assert(v1.type == STRING || v1.type == NAME);
+    assert(b1.type == STRING);
+    assert(e.type == STRING || e.type == NUM);
+    assert(snd.type == STRING);
+    assert(i.type == STRING);
+
+    Common::String *bmp = new Common::String(i.u.str);
+    assert(g_private->isDemo() || strcmp(bmp->c_str(), "\"\"") != 0);
+
+    if (v1.type == STRING)
+        assert(strcmp(v1.u.str, "\"\"") == 0);
+
+    debugC(1, kPrivateDebugScript, "Inventory(...)");
+    if (strcmp(b1.u.str, "\"\"") != 0) {
+        Common::String *s = new Common::String(b1.u.str);
+        MaskInfo *m = (MaskInfo *)malloc(sizeof(MaskInfo));
+        m->surf = g_private->loadMask(*s, 0, 0, true);
+
+        if (e.type == NUM)
+            m->nextSetting = NULL;
+        else
+            m->nextSetting = new Common::String(e.u.str);
+
+        m->cursor = new Common::String("kInventory");
+        m->point = new Common::Point(0,0);
+
+        if (v1.type == NAME)
+            m->flag1 = v1.u.sym;
+        else
+            m->flag1 = NULL;
+
+        if (v2.type == NAME)
+            m->flag2 = v2.u.sym;
+        else
+            m->flag2 = NULL;
+
+        g_private->_masks.push_front(*m);
+        g_private->_toTake = true;
+
+        Common::String *f;
+        if (strcmp(snd.u.str, "\"\"") != 0) {
```

Comment:
Looks like you could review usage of `strcmp` and if it refers to `.u.str`, just replace it with a direct comparison.

Reply:
Fixed.